QueueProcessors/QueueProcessor/queue_processor.py

```diff
@@ -424,96 +427,31 @@ def retry_run(reduction_run, retry_in):
-        self._use_ssl = use_ssl
-
-    def set_listener(self, listener):
-        """ Set the listener class. """
-        self._listener = listener
-
-    def get_connection(self, listener=None):
-        """ Return the connection and connect if not already connected. """
-        if listener is None and not self._client_only:
-            if self._listener is None:
-                listener = Listener(self)
-                self._listener = listener
-            else:
-                listener = self._listener
-
-        logger.info("[%s] Connecting to %s", self._consumer_name, str(self._brokers))
-
-        connection = stomp.Connection(host_and_ports=self._brokers, use_ssl=self._use_ssl)
-        if not self._client_only:
-            connection.set_listener(self._consumer_name, listener)
-        connection.start()
-        connection.connect(self._user, self._password, wait=False)
-
-        time.sleep(0.5)
-        return connection
-
-    def connect(self):
-        """ Connect and subscribe to the queues. """
-        if self._connection is None or not self._connection.is_connected():
-            self._disconnect()
-            self._connection = self.get_connection()
-
-        for queue in self._topics:
-            logger.info("[%s] Subscribing to %s", self._consumer_name, queue)
-            self._connection.subscribe(destination=queue, id=1, ack='auto')
-
-    def _disconnect(self):
-        """ Disconnect form the server. """
-        if self._connection is not None and self._connection.is_connected():
-            self._connection.disconnect()
-        self._connection = None
-        logger.info("[%s] Disconnected", self._consumer_name)
-
-    def stop(self):
-        """ Stop the connection. """
-        self._disconnect()
-        if self._connection is not None:
-            self._connection.stop()
-        self._connection = None
-
-    def send(self, destination, message, persistent='true', priority='4', delay=None):
-        """ Send a message to a queue. """
-        if self._connection is None or not self._connection.is_connected():
-            self._disconnect()
-            self._connection = self.get_connection()
-
-        headers = {}
-        if delay:
-            headers['AMQ_SCHEDULED_DELAY'] = str(delay)
-        self._connection.send(destination, message, persistent=persistent, priority=priority,
-                              headers=headers)
-        logger.debug("[%s] send message to %s", self._consumer_name, destination)
-
+def setup_connection(consumer_name):
+    """ Starts the ActiveMQ connection and registers the event listener """
+    logger.info("Starting autoreduce queue connection")
+    # Connect to ActiveMQ
+    activemq_client = QueueClient()
+    activemq_client.connect()
+
+    # Register the event listener
+    conn = activemq_client.get_connection()    
+    listener = Listener(activemq_client)
+    conn.set_listener(consumer_name, listener)
+    time.sleep(0.5)
```

Comment:
This sleep is not required here as it is now performed in the `QueueClient` [here](https://github.com/ISISScientificComputing/autoreduce/blob/master/utils/clients/queue_client.py#L57) as such it can be deleted.